test/sqllogictest/autogenerated_all_parts_essential.slt

```diff
@@ -0,0 +1,994 @@
+1992-04-14  116216  4  1
+1992-04-14  141741  13  3
+1992-04-14  143368  4  7
+1992-04-15  18459  14  4
+1992-04-15  21937  7  6
+1992-04-15  26240  8  2
+1992-04-15  70493  5  5
+1992-04-15  115297  4  1
+1992-04-15  140822  13  3
+1992-04-15  142449  4  7
+1992-04-18  -15875  14  4
+1992-04-18  -12397  7  6
+1992-04-18  -8094  8  2
+1992-04-18  36159  5  5
+
+query T multiline
+EXPLAIN SELECT l_receiptDATE , o_totalprice - l_extendedprice , o_custkey , o_orderkey FROM lineitem JOIN orders ON ( l_shipDATE != o_orderdate ) WHERE l_shipDATE BETWEEN '1993-01-01' AND '1993-01-28' AND l_shipDATE > o_orderdate OR o_orderkey <= 22 ORDER BY 1 , 2 , 3 , 4 LIMIT 25
+----
+%0 =
+| Get materialize.public.lineitem (u9)
+| Project (#5, #10, #12)
+| ArrangeBy ()
+
+%1 =
+| Get materialize.public.orders (u5)
+| Project (#0, #1, #3, #4)
+
+%2 =
+| Join %0 %1
+| | implementation = Differential %1 %0.()
+| Filter ((((#1 <= 1993-01-28) && (#1 >= 1993-01-01)) && (#1 > #6)) || (#3 <= 22)), (#1 != #6)
+| Map (#5 - #0)
+| Project (#2, #7, #4, #3)
+
+Finish order_by=(#0 asc, #1 asc, #2 asc, #3 asc) limit=25 offset=0 project=(#0..#3)
+
+EOF
+
+
+query IT
+SELECT o_orderkey , l_receiptDATE FROM lineitem LEFT JOIN orders ON ( l_commitDATE = o_orderdate ) WHERE l_commitDATE >= '1998-03-22' AND l_orderkey BETWEEN 59 AND 95 + 5 OR l_receiptDATE = o_orderdate
+----
+NULL  1998-03-30
+NULL  1998-03-31
+NULL  1998-04-11
+NULL  1998-05-22
+NULL  1998-06-02
+NULL  1998-06-03
+NULL  1998-06-07
+NULL  1998-06-29
+NULL  1998-06-29
+NULL  1998-07-02
+NULL  1998-07-05
+NULL  1998-07-06
+NULL  1998-07-21
+NULL  1998-08-14
+NULL  1998-08-29
+390  1998-02-18
+
+query T multiline
+EXPLAIN SELECT o_orderkey , l_receiptDATE FROM lineitem LEFT JOIN orders ON ( l_commitDATE = o_orderdate ) WHERE l_commitDATE >= '1998-03-22' AND l_orderkey BETWEEN 59 AND 95 + 5 OR l_receiptDATE = o_orderdate
+----
+%0 =
+| Get materialize.public.lineitem (u9)
+| Project (#0, #11, #12)
+| ArrangeBy (#1)
+
+%1 =
+| Get materialize.public.orders (u5)
+| Project (#0, #4)
+
+%2 = Let l0 =
+| Join %0 %1 (= #1 #4)
+| | implementation = Differential %1 %0.(#1)
+| Project (#0..#3)
+
+%3 = Let l1 =
+| Get materialize.public.lineitem (u9)
+| Filter (null || (((#0 <= 100) && (#0 >= 59)) && (#11 >= 1998-03-22)))
+
+%4 =
+| Get %3 (l1)
+| Project (#11, #12)
+
+%5 =
+| Get %2 (l0)
+| Project (#1)
+| Distinct group=(#0)
+| ArrangeBy (#0)
+
+%6 =
+| Join %4 %5 (= #0 #2)
+| | implementation = Differential %4 %5.(#0)
+| Project (#1)
+| Negate
+
+%7 =
+| Get %3 (l1)
+| Project (#12)
+
+%8 =
+| Union %6 %7
+| Map null
+| Project (#1, #0)
+
+%9 =
+| Get %2 (l0)
+| Filter ((((#0 <= 100) && (#0 >= 59)) && (#1 >= 1998-03-22)) || (#1 = #2))
+| Project (#3, #2)
+
+%10 =
+| Union %8 %9
+
+EOF
+
+query II
+SELECT c_custkey , o_custkey FROM lineitem JOIN orders ON ( l_orderkey = o_orderkey ) JOIN customer ON ( c_acctbal <= o_totalprice ) WHERE l_quantity = 17 AND c_comment BETWEEN 's' AND 'x' OR l_quantity  IN ( 7 , 33 )
+----
+1  1
```

Comment:
Expected results should be in "values hashing to" form. See https://github.com/MaterializeInc/materialize/pull/8392#issuecomment-925299207

Reply:
Yes, thanks for reminding me that this feature exists -- I was struggling with some queries that returned large result sets.

Reply:
Done, plus a fix to support hashed results with --rewrite-results in https://github.com/MaterializeInc/materialize/pull/9998